analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseTrueForAll.cs

```diff
@@ -0,0 +1,123 @@
+        var nullableMethodGood = DoWork()?.TrueForAll(x => true); // Compliant
+
+        var inlineInitialization = new List<int> { 42 }.All(x => true); // Noncompliant
+
+        var imposter = new ContainsAllMethod<int>();
+        imposter.All(x => true); // Compliant
+
+        var badList = new BadList<int>();
+        badList.All(x => true); // Compliant
+
+        var goodList = new GoodList<int>();
+        goodList.All(x => true); // Noncompliant
+
+        var ternary = (true ? list : goodList).All(x => true); // Noncompliant
+        var nullCoalesce = (list ?? goodList).All(x => true); // Noncompliant
+        var ternaryNullCoalesce = (list ?? (true ? list : goodList)).All(x => true); // Noncompliant
+
+        _ = goodList.Fluent().Fluent().Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent().Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent().Fluent()?.Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent().Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent().Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent().Fluent()?.All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent()?.Fluent().All(x => true); //Noncompliant
+        _ = goodList.Fluent()?.Fluent()?.Fluent()?.Fluent()?.All(x => true); //Noncompliant
+        //                                                   ^^^
+
+        All<int>(x => true); // Compliant
+        AcceptMethod<int>(goodList.All); //FP this is not an invocation, just a member access
+    }
+
+    bool All<T>(Func<T, bool> predicate) => true;
+    void AcceptMethod<T>(Func<Func<T, bool>, bool> methodThatLooksLikeAll) { }
+
+    class GoodList<T> : List<T>
+    {
+        public GoodList<T> Fluent() => this;
+
+        void CallAll() =>
+            this.All(x => true); // Noncompliant
+    }
+
+    class BadList<T> : List<T>
+    {
+        public bool All(Func<T, bool> predicate) => true;
+    }
+
+    class ContainsAllMethod<T>
+    {
+        public bool All(Func<T, bool> predicate) => true;
+    }
+}
+
+class ArrayTestcases
```

Comment:
I think we could split test case files per type. Having 3 in total: one for `List`, one for `ImmutableList`, and one for `Array`.

Reply:
I don't think that this provides any value, usually the tests are split between .NET versions and the libraries you are using. I have put the Array and List testcases in classes so you can easily collapse/expand them. See the `#if NET` comment for more details.